arches/app/models/resource.py

```diff
@@ -662,7 +663,16 @@ def get_relations(resourceinstanceid, start, limit, resourceinstance_graphid=Non
             resourceinstance_graphid=resourceinstance_graphid,
         )
 
+        resource_relations["relations"] = list(
+            filter(
+                lambda x: user_can_read_resource(user, x.resourceinstanceidto_id)
+                and user_can_read_resource(user, x.resourceinstanceidfrom_id),
+                resource_relations["relations"],
+            )
+        )
+        resource_relations["total"] = len(resource_relations["relations"])
```

Comment:
This ignores pagination and would set the `total` value to an inaccurate result. Unfortunately I don't see a fantastic way to obtain an accurate `total` without altering the queries on L644-L650